mtags/src/main/scala-3/scala/meta/internal/pc/SemanticTokenProvider.scala

```diff
@@ -0,0 +1,207 @@
+
+    def filterOutFlags(sym: Symbol): Boolean =
+      sym.is(Flags.Synthetic) ||
+        sym.is(Flags.ImplicitVal) ||
+        sym.owner.is(Flags.ImplicitVal)
+
+    candidates
+      .takeWhile(_.pos.start == tk.pos.start)
+      .filter(isTarget) match
+      case Nil => None
+      case node :: Nil =>
+        Some((node, candidates))
+      case manyNodes =>
+        val preFilter = manyNodes.filter { case NodeInfo(sym, p) =>
+          Collector
+            .symbolAlternatives(sym)
+            .exists(_.decodedName == tk.text)
+        }
+        preFilter match
+          case Nil => manyNodes.headOption.map(ni => (ni, candidates))
+          case matchingNames
+              if matchingNames.exists(ni =>
+                !filterOutOwner(ni.sym) && !filterOutFlags(ni.sym)
+              ) =>
+            matchingNames.collectFirst {
+              case ni if !filterOutOwner(ni.sym) && !filterOutFlags(ni.sym) =>
+                (ni, candidates)
+            }
+          case head :: _ => Some((head, candidates))
+        end match
+    end match
+  end pickFromTraversed
+
+  /**
+   * returns (SemanticTokenType, SemanticTokenModifier) of @param tk
+   */
+  override def identTypeAndMod(
+      ident: Token.Ident,
+      nodesIterator: List[NodeInfo],
+  ): (Int, Int, List[NodeInfo]) =
+    val default = (-1, 0, nodesIterator)
+
+    val ret =
+      for (
+        (nodeInfo, nodesIterator) <- pickFromTraversed(ident, nodesIterator);
+        sym = nodeInfo.sym
+      ) yield
+
+        var mod: Int = 0
+        def addPwrToMod(tokenID: String) =
+          val place: Int = getModifierId(tokenID)
+          if place != -1 then mod += (1 << place)
+        // get Type
+        val typ =
+          if sym.is(Flags.Param) && !sym.isTypeParam
+          then getTypeId(SemanticTokenTypes.Parameter)
+          else if sym.isTypeParam || sym.isSkolem then
+            getTypeId(SemanticTokenTypes.TypeParameter)
+          else if isOperatorName(ident) then
+            getTypeId(SemanticTokenTypes.Operator)
+          else if sym.is(Flags.Enum) || sym.isAllOf(Flags.EnumVal)
+          then getTypeId(SemanticTokenTypes.Enum)
+          else if sym.is(Flags.Trait) then
+            getTypeId(SemanticTokenTypes.Interface) // "interface"
+          else if sym.isClass then
+            getTypeId(SemanticTokenTypes.Class) // "class"
+          else if sym.isType && !sym.is(Flags.Param) then
+            getTypeId(SemanticTokenTypes.Type) // "type"
+          else if sym.is(Flags.Mutable) then
+            getTypeId(SemanticTokenTypes.Variable) // "var"
+          else if sym.is(Flags.Package) then
+            getTypeId(SemanticTokenTypes.Namespace) // "package"
+          else if sym.is(Flags.Module) then
+            getTypeId(SemanticTokenTypes.Class) // "object"
+          else if sym.is(Flags.Method) then
+            if sym.isGetter | sym.isSetter then
+              getTypeId(SemanticTokenTypes.Variable)
```

Comment:
I'm not sure about this case.
In Scala 3 `<<List>>(1,2,3)` gets here, in Scala 2 it becomes `/*variable,readonly*/`. Maybe we should somehow treat it as `class`?